pkg/cloud/cloud.go

```diff
@@ -311,6 +311,7 @@ func (c *cloud) DetachDisk(ctx context.Context, volumeID, nodeID string) error {
 
 	if !device.IsAlreadyAssigned {
 		glog.Warningf("DetachDisk called on non-attached volume: %s", volumeID)
+		return nil
```

Comment:
The in-tree implementation doesn't return early, and I think there's a reason for that: the `device` object isn't really synchronized with the actual state of the world.

However, since `GetDevice()` loads all devices in use by the instance, I don't think there would be a problem here. 

I think the safest option would be to call `DetachVolumeWithContext` and wait for the volume state. However, I don't think this patch is incorrect, so I'm OK with merging it.

We might also want to take the chance and `Release()` the device only if

Reply:
Ah. Let me take some time to read through the in-tree implementation first and understand more.